schemas/telemetry/configuration-schema.json

```diff
@@ -24,6 +24,7 @@
             "configuration": {
               "type": "object",
               "description": "Configuration properties",
+              "additionalProperties": true,
```

Comment:
Why `additionalProperties: true` 🤔💭? Do we want to allow arbitrary configuration fields?

Reply:
What is the impact of reporting arbitrary fields?
Are they going to be treated just like the ones that are declared?

I find it convenient for temporary fields, when we want to monitor deprecated fields usage for instance, or when we want to deprecate certain versions of a dependency (like React Native or React).
The risk of adding those in the schema is that we forget to remove them and it becomes bloated after a few months, and it also increases a little the time to be able to report it.

Reply:
I just wonder where is the line between reporting something as a custom property vs tracking it in the schema. If it's not about "platform-specifics" (we have Flutter- or Android-specific fields), then what else? The example of deprecated configuration makes sense and I like it 👍, but I believe that using `additionalProperties` should be followed with some agreed convention, to avoid ambiguity.